swap/tests/testutils/mod.rs

```diff
@@ -28,12 +28,12 @@ pub struct StartingBalances {
 pub struct Test {
     swap_amounts: SwapAmounts,
 
-    alice_swap_factory: Option<alice::SwapFactory>,
+    alice_swap_factory: Option<alice::Builder>,
```

Comment:
Agree with the rename if it is only useful once. However, having an `Option` so you can `take` it is seems kind of weird.

You could implement `Clone` on the builder to make new instances without having to `take` it.

Also, this `Test` struct seems to be doing way too many things. Why not just create more entrypoins on `testutils` for the different kinds of tests there are? You can then customize the parameters that are passed to the closure based on the different test scenarios (happy path, resumable, etc). That would drive down the complexity of this struct and save you LoC in the test because you directly receive the arguments you need.

I prefer duplication in tests if it helps readability. A test is worth nothing if I cannot judge whether it is testing the right thing :)